transport/packet/src/v1.rs

```diff
@@ -1,173 +1,120 @@
-use std::{
-    fmt::Formatter,
-    ops::{Add, Range, Sub},
-};
+use std::{fmt::Formatter, ops::Range};
 
-/// Tags are represented by 8 bytes ([`u64`]`).
-///
-/// [`u64`] should offer enough space to even avoid collisions and tag attacks.
-#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Tag(pub u64);
-
-/// The tag appended to the application data packet to identify the application target.
-impl Tag {
-    pub const MAX: Tag = Tag(u64::MAX);
-    pub const SIZE: usize = size_of::<u64>();
-    /// Tag range usable by external applications of the library.
-    pub const USABLE_RANGE: Range<Tag> = ReservedTag::UPPER_BOUND..CustomTag::UPPER_BOUND;
-
-    pub const fn new(tag: u64) -> Self {
-        Self(tag)
-    }
-
-    pub fn from_be_bytes(bytes: [u8; Self::SIZE]) -> Self {
-        Self(u64::from_be_bytes(bytes))
-    }
+use strum::IntoEnumIterator;
 
-    pub fn to_be_bytes(&self) -> [u8; Self::SIZE] {
-        self.0.to_be_bytes()
-    }
-}
-
-impl serde::Serialize for Tag {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        serializer.serialize_u64(self.0)
-    }
-}
-
-impl<'a> serde::Deserialize<'a> for Tag {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'a>,
-    {
-        let value = u64::deserialize(deserializer)?;
-        Ok(Self(value))
-    }
-}
+/// List of all reserved application tags for the protocol.
+#[repr(u64)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, strum::EnumIter)]
+pub enum ReservedTag {
+    /// Request to initiate a new session.
+    SessionStart = 0,
+    /// Confirmation that a new session has been established by the counterparty.
+    SessionEstablished = 1,
+    /// Counterparty could not establish a new session due to an error.
+    SessionError = 2,
+    /// Counterparty has closed the session.
+    SessionClose = 3,
+    /// A ping message to keep the session alive.
```

Comment:
Please, clean this up, only the general tag for `Session` or `SessionStart` should be present, maybe at tag `1`. The probing mechanism can be tag `0`.